master/internal/resourcemanagers/resource_pool.go

```diff
@@ -381,24 +380,30 @@ func (rp *ResourcePool) receiveAgentMsg(ctx *actor.Context) error {
 	return nil
 }
 
-func (rp *ResourcePool) moveJob(msg job.MoveJob) error {
-	// REMOVEME
-	fmt.Println(rp.config.PoolName, "moveJob: ", msg)
+func (rp *ResourcePool) moveJob(msg job.MoveJob) (job.RegisterJobPosition, error) {
 	rp.queuePositions[job.TailAnchor] = initalizeQueuePosition(time.Now())
-	newPos, rebalance, err := computeNewJobPos(msg, rp.queuePositions)
+	newPos, err := computeNewJobPos(msg, rp.queuePositions)
 	if err != nil {
-		return err
+		return job.RegisterJobPosition{}, err
 	}
 	rp.queuePositions[msg.ID] = newPos
-	// condiontally check if rebalancing is needed.
-	if rebalance {
-		adjustPriorities(&rp.queuePositions)
-	}
 
-	if err := rp.persist(); err != nil {
+	// actors.NotifyAfter(ctx, 0, schedulerTick{}) ?
```

Comment:
I think this was Hamid's comment (I forked off of his initial work). I forgot to ask about this, but it might not be a great idea, especially if we're handling bulk operations.